node/src/utils/pidfile.rs

```diff
@@ -0,0 +1,255 @@
+//! Pidfile utility type.
+//!
+//! Pidfiles are used to gate access to a resource, as well as detect unclean shutdowns.
+
+use std::{
+    fs::{self, File},
+    io::{self, Read, Seek, SeekFrom, Write},
+    num::ParseIntError,
+    path::{Path, PathBuf},
+    process,
+};
+
+use fs2::FileExt;
+use thiserror::Error;
+use tracing::warn;
+
+/// A PID file.
+///
+/// Records the current process' PID, removes it on exit. Can be used to determine whether or not
+/// an application was shut down cleanly.
+///
+/// The pidfile is held open with an exclusive but advisory lock.
+#[derive(Debug)]
+pub struct Pidfile {
```

Comment:
```suggestion
pub struct PidFile {
```

I'd like to see the module renamed too please (to `pid_file`).

Reply:
Can do. Thanks, rust-analyzer!